datafusion-examples/examples/advanced_udf.rs

```diff
@@ -191,6 +199,51 @@ impl ScalarUDFImpl for PowUdf {
     }
 }
 
+/// Evaluate `base ^ exp` *without* allocating a new array, if possible
+fn pow_in_place(base: f64, exp_array: ArrayRef) -> Result<ArrayRef> {
```

Comment:
so it's "pow maybe in place", not "pow in place"
worth reflecting in the function name (eg with addition of "maybe")? otherwise i'd call this just `pow`